cdpctl/validation/aws_utils.py

```diff
@@ -198,11 +195,11 @@ def get_role(
     try:
         role = iam_client.get_role(RoleName=role_name)
     except iam_client.exceptions.NoSuchEntityException:
-        pytest.fail(role_missing_message.format(role_name), False)
+        fail(template=missing_issue, resources=[role_name])
     except iam_client.exceptions.ServiceFailureException as e:
-        raise Exception(service_failure_message) from e
+        raise UnrecoverableValidationError(service_failure_message) from e
     # handling stub client error
-    except ClientError:
-        pytest.fail(role_missing_message.format(role_name), False)
+    except ClientError as ce:
+        fail(missing_issue, ce.args[0])
```

Comment:
why not use `role_name` here instead of ce.args? What does that look like?

Reply:
Is this always a comm error? Hence no role in error?

Reply:
yeah mistype